cirq/devices/device.py

```diff
@@ -12,20 +12,44 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import TYPE_CHECKING
-
 import abc
-
+from typing import TYPE_CHECKING, Optional, AbstractSet
 
 if TYPE_CHECKING:
     import cirq
 
-# Note: circuit/schedule types specified by name to avoid circular references.
-
 
 class Device(metaclass=abc.ABCMeta):
     """Hardware constraints for validating circuits."""
 
+    def qubit_set(self) -> Optional[AbstractSet['cirq.Qid']]:
+        """Returns a set or frozenset of qubits on the device, if possible.
+
+        This method returns None when the set of qubits is unavailable. For
+        example, `cirq.UnconstrainedDevice` returns `None` because allows any
```

Comment:
This is redundant with the Returns statement.  I would probably keep one or the other rather than repeating it.

Reply:
Done.